### ISSUE TITLE
Lua language server refused to load this directory(home directory)

### ISSUE THREAD

**Issue by ParsaNobahari** [siduck76/NvChad#2177]
Hi.

So whenever I open a .lua file in some directories it give me a warning:
> LSP[lua_ls] Your workspace is set to `/home/parsa`. Lua language server refuse
d to load this directory. Please check your configuration.[learn more here](ht
tps://github.com/LuaLS/lua-language-server/wiki/FAQ#why-is-the-server-scanning
-the-wrong-folder)

I went to the [link](https://github.com/LuaLS/lua-language-server/wiki/FAQ#why-is-the-server-scanning-the-wrong-folder) and kinda understand why this is happening. I think [this person](https://github.com/folke/neodev.nvim/issues/50#issuecomment-1237137332) got the closest to fixing the issue, he also made a pr but the issue exists for me. I took a look at [neovim doc](https://neovim.io/doc/user/lsp.html) but I didn't find any way to change the directory that gets reported from client to lsp. I thought maybe I can find a way to stop neovim sending uri of the home directory but no luck there either.

**Unexpected behaviors**
So when I open ~/.config/nvim/init.lua I don't get the error. My best guess is because of the other files in the directory which are:
- .luarc.json
- selene.toml
- .stylua.toml

I tried copying these files to ~ directory but warning remains.

**Workspace:**
 - Archlinux
 - Alacritty
 - NVIM v0.9.1
 - lua-language-server v3.6.22

I also couldn't find the server log file anywhere mentioned [here](https://github.com/LuaLS/lua-language-server/wiki/FAQ#where-can-i-find-the-log-file) which is suggested to be at _lua-language-server/log/_ I don't know where this suppose to be -_-
Also that [same site](https://github.com/LuaLS/lua-language-server/wiki/FAQ#how-do-i-fix-this) suggests to:
> check the rootUri field in your configuration - some clients allow users to customize this value.

Again you can guess I couldn't find where I can do this so I assumed I should create an issue.
Honestly I'm not even sure if this is the right place for creating one.

Thanks for your patience

best regards

**Comment by siduck** [comment]
@ParsaNobahari please check lua-ls section in lspconfig's server_configurations.md file, this   issue has nothing to do with nvchad!

maybe you can change root_dir to the cwd

**Comment by ParsaNobahari** [comment]
If anyone ever wonder, I changed from

`root_dir = lspconfig.util.root_pattern(".luarc.json", ".luarc.jsonc",
        ".luacheckrc", ".stylua.toml", "stylua.toml", "selene.toml",
        "selene.yml", ".git")`

to

`lspconfig.lua_ls.setup({
    cmd = {"lua-language-server"},
    filetypes = {"lua"},
    root_dir = lspconfig.util.root_pattern(
        "~/.local/share/nvim/lazy/nvim-lspconfig/.luarc.json",
        "~/.local/share/nvim/lazy/nvim-lspconfig/.luarc.jsonc",
        "~/.local/share/nvim/lazy/nvim-lspconfig/.luacheckrc",
        "~/.local/share/nvim/lazy/nvim-lspconfig/.stylua.toml",
        "~/.local/share/nvim/lazy/nvim-lspconfig/stylua.toml",
        "~/.local/share/nvim/lazy/nvim-lspconfig/selene.toml",
        "~/.local/share/nvim/lazy/nvim-lspconfig/selene.yml",
        "~/.local/share/nvim/lazy/nvim-lspconfig/.git"),
    single_file_support = true
})
`

Basically adding `~/.local/share/nvim/lazy/nvim-lspconfig/` to the beginning of each one file

and the warning is gone

**Comment by ianTevesAcc** [comment]
This is how I fixed mine.

```lua
-- Load defaults for Lua language server
require("nvchad.configs.lspconfig").defaults()

local lspconfig = require "lspconfig"
local nvlsp = require "nvchad.configs.lspconfig"

-- Lua language server configuration
lspconfig.lua_ls.setup {
  on_attach = nvlsp.on_attach,
  on_init = nvlsp.on_init,
  capabilities = nvlsp.capabilities,
  cmd = { "lua-language-server" },
  filetypes = { "lua" },
  root_dir = function(fname)
    -- Try to find a proper root using Git or Lua config files
    local root = lspconfig.util.find_git_ancestor(fname)
    if root then
      return root -- If Git repository exists, use it as the root
    end
    -- Otherwise, fallback to finding Lua configuration files
    local root_pattern = lspconfig.util.root_pattern(
      ".luarc.json",
      ".luarc.jsonc",
      ".luacheckrc",
      ".stylua.toml",
      "stylua.toml",
      "selene.toml",
      "selene.yml",
      ".git"
    )
    -- If no root config found, fallback to current working directory
    return root_pattern(fname) or vim.fn.getcwd()
  end,
  single_file_support = true, -- Support for single Lua files
}
```
